File: internship_backend_nov24/jury_solutions/cs/d.cs
using System;
using System.Linq;


class Program
{
    static void Main()
    {
        Solve();
    }

    public static void Solve()
    {
        var bounds = Console.ReadLine().Split().Select(long.Parse).ToArray();
        var maxX = bounds[0];
        var maxY = bounds[1];
        
        var pointsCount = Console.ReadLine().Split().Select(long.Parse).ToArray();
        var xCount = pointsCount[0];
        var yCount = pointsCount[1];
        
        var xLines = Console.ReadLine().Split().Select(long.Parse).ToList();
        var yLines = Console.ReadLine().Split().Select(long.Parse).ToList();
        
        var queriesCount = int.Parse(Console.ReadLine());
        
        xLines.Add(0);
        xLines.Add(maxX);
        xLines.Sort();
        
        yLines.Add(0);
        yLines.Add(maxY);
        yLines.Sort();

        while (queriesCount-- > 0)
        {
            var query = Console.ReadLine().Split().Select(long.Parse).ToArray();
            var x1 = query[0];
            var y1 = query[1];
            
            var x2 = query[2];
            var y2 = query[3];

            var xRangDiff = xLines.BinarySearch(x1) - xLines.BinarySearch(x2) == 0;
            var yRangDiff = yLines.BinarySearch(y1) - yLines.BinarySearch(y2) == 0;
            
            if (xRangDiff && yRangDiff)
                Console.WriteLine("YES");
            else
                Console.WriteLine("NO");
        }
        
    }
}

File: internship_backend_nov24/jury_solutions/python/d.py
from bisect import bisect_left

maxX, maxY = map(int, input().split())
xCount, yCount = map(int, input().split())
xLines = list(map(int, input().split()))
yLines = list(map(int, input().split()))
queriesCount = int(input())
xLines.append(0)
xLines.append(maxX)
yLines.append(0)
yLines.append(maxY)
xLines.sort()
yLines.sort()
for _ in range(queriesCount):
    x1, y1, x2, y2 = map(int, input().split())
    diffX = bisect_left(xLines, x1) == bisect_left(xLines, x2)
    diffY = bisect_left(yLines, y1) == bisect_left(yLines, y2)
    if diffX and diffY:
        print("YES")
    else:
        print("NO")
